File: generator/src/test/resources/kotlin/expected/sunday/RequestBodyParamTest/test-generation-of-body-parameter-with-explicit-content-type.output.kt
package io.test.service

import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.Operation
import io.outfoxx.sunday.OperationSpec
import io.outfoxx.sunday.Transport
import io.outfoxx.sunday.http.Method
import io.outfoxx.sunday.http.Request
import io.outfoxx.sunday.operation
import kotlin.Any
import kotlin.ByteArray
import kotlin.String
import kotlin.collections.List
import kotlin.collections.Map

public class API<Req : Request>(
  public val transport: Transport<Req>,
  public val defaultContentTypes: List<MediaType> = listOf(),
  public val defaultAcceptTypes: List<MediaType> = listOf(MediaType.JSON),
) {
  public fun fetchTest(body: ByteArray): Operation<ByteArray, Map<String, Any>, Req> =
      this.transport.operation<ByteArray, Map<String, Any>, Req>(
    OperationSpec(
      method = Method.Get,
      pathTemplate = "/tests",
      body = body,
      contentTypes = listOf(MediaType.OctetStream),
      acceptTypes = this.defaultAcceptTypes
    )
  )
}

File: generator/src/test/resources/swift/expected/RequestBodyParamTest/test-generation-of-body-parameter-with-explicit-content-type.output.swift
import Foundation
import PotentCodables
import Sunday

public class API {

  public let requestFactory: RequestFactory
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    requestFactory: RequestFactory,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = [.json]
  ) {
    self.requestFactory = requestFactory
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
  }

  public func fetchTest(body: Data) async throws -> [String : AnyValue] {
    return try await self.requestFactory.result(
      method: .get,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: nil,
      body: body,
      contentTypes: [.octetStream],
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

}
